packages/gestalt/src/Typeahead.js

```diff
@@ -72,24 +72,18 @@ const TypeaheadWithForwardRef: React$AbstractComponent<Props, HTMLInputElement>
     zIndex,
   } = props;
 
-  // Store original data
-  const dataRef = useRef(options);
```

Comment:
This was the source of the bug, since the ref was initiated with the initial `options` value but then persisted, so would never update.